extensions/transaction/transaction-local/src/main/java/org/eclipse/dataspaceconnector/transaction/local/LocalTransactionContext.java

```diff
@@ -59,26 +59,32 @@ public void execute(TransactionBlock block) {
             }
             throw new EdcException(e.getMessage(), e);
         } finally {
-            if (startedTransaction) {
-                if (transaction.isRollbackOnly()) {
-                    resources.forEach(localTransactionResource -> {
-                        try {
-                            localTransactionResource.rollback();
-                        } catch (Exception e) {
-                            monitor.severe("Error rolling back resource", e);
-                        }
-                    });
-                } else {
-                    resources.forEach(localTransactionResource -> {
-                        try {
-                            localTransactionResource.commit();
-                        } catch (Exception e) {
-                            monitor.severe("Error committing resource", e);
-                        }
-                    });
-                }
-                transactions.remove();
+            commitOrRollback(startedTransaction, transaction);
+        }
+    }
+
+    @Override
+    public <T> T execute(ResultTransactionBlock<T> block) {
```

Comment:
A good way to refactor this and avoid all the duplication would be to make the "no result execute" call the "result execute" making it return null, like:
```java
    @Override
    public void execute(TransactionBlock block) {
        execute((ResultTransactionBlock<Void>) () -> {
            block.execute();
            return null;
        });
    }
```